hydrator-common/src/main/java/io/cdap/plugin/common/LineageRecorder.java

```diff
@@ -85,23 +85,25 @@ public void createExternalDataset(@Nullable Schema schema) {
    * @param fields output fields of this read operation
    */
   public void recordRead(String operationName, String operationDescription, List<String> fields) {
-    context.record(Collections.singletonList(new FieldReadOperation(operationName,
-                                                                    operationDescription,
-                                                                    EndPoint.of(context.getNamespace(), dataset),
-                                                                    fields)));
+    fields.forEach(field -> context.record(
+      Collections.singletonList(new FieldReadOperation(operationName + field,
+                                                       operationDescription,
+                                                       EndPoint.of(context.getNamespace(), dataset),
+                                                       field))));
   }
 
   /**
    * Records a {@link WriteOperation}
    *
    * @param operationName the name of the operation
    * @param operationDescription description for the operation
-   * @param fields input fields of this read operation
+   * @param fields input fields of this write operation
    */
   public void recordWrite(String operationName, String operationDescription, List<String> fields) {
-    context.record(Collections.singletonList(new FieldWriteOperation(operationName,
-                                                                     operationDescription,
-                                                                     EndPoint.of(context.getNamespace(), dataset),
-                                                                     fields)));
+    fields.forEach(field -> context.record(
```

Comment:
It is confusing to me why one of these is 'correct' and the other is not. Conceptually, it makes sense to me that the sink is performing a single operation. Why does it need to be an operation per field?